File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        # nums.sort()
        # return nums[math.floor(len(nums)/2)]

        a = nums[0]
        counter = 0

        for i in nums:
            if i == a:
                counter+=1
            else:
                counter-=1
            
            if counter == 0:
                a = i
                counter = 1
        
        return a

File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    var a = nums[0]
    var counter = 0

    for (num of nums){
        if(a === num){
            counter++
        }else{
            counter--
        }

        if(counter === 0){
            counter = 1
            a = num
        }
    }

    return a
};